core/src/lib.rs

```diff
@@ -145,89 +187,101 @@ mod tests {
-        if type_rs.is_ok() {
-            kv_subject
-                .entry("type".to_string())
-                .or_insert(type_rs.unwrap());
-        }
-
-        kv_subject
-            .entry("givenName".to_string())
-            .or_insert(Value::String("JOHN".to_string()));
-        kv_subject
-            .entry("familyName".to_string())
-            .or_insert(Value::String("SMITH".to_string()));
-        kv_subject
-            .entry("gender".to_string())
-            .or_insert(Value::String("Male".to_string()));
-        kv_subject
-            .entry("image".to_string())
-            .or_insert(Value::String(
-                "data:image/png;base64,iVBORw0KGgo...kJggg==".to_string(),
-            ));
-        kv_subject
-            .entry("residentSince".to_string())
-            .or_insert(Value::String("2015-01-01".to_string()));
-        kv_subject
-            .entry("lprCategory".to_string())
-            .or_insert(Value::String("C09".to_string()));
-        kv_subject
-            .entry("lprNumber".to_string())
-            .or_insert(Value::String("999-999-999".to_string()));
-        kv_subject
-            .entry("commuterClassification".to_string())
-            .or_insert(Value::String("C1".to_string()));
-        kv_subject
-            .entry("birthCountry".to_string())
-            .or_insert(Value::String("Bahamas".to_string()));
-        kv_subject
-            .entry("birthDate".to_string())
-            .or_insert(Value::String("1958-07-17".to_string()));
+        let (kv_body, kv_subject) = get_body_subject();
 
         let vc = to.create_credential(
             crate::CRED_TYPE_PERMANENT_RESIDENT_CARD.to_string(),
             kv_subject,
             kv_body,
             "https://issuer.oidp.uscis.gov/credentials/83627465",
         );
+
         assert!(vc.is_ok());
-        assert_json_eq!(expect, vc.unwrap());
+        let credential = vc.unwrap();
+        assert_json_eq!(expect_credential, credential.serialize());
+
+        let signer = signature::signer::Ed25519DidSigner::new();
+        let proof = create_data_integrity_proof(
+            &signer,
+            credential.serialize(),
+            signature::suite::VerificationRelation::AssertionMethod,
+        );
+
+        assert!(proof.is_ok());
+
+        let verifiable_credential = credential.create_verifiable_credentials(proof.unwrap());
+        let credentials = vec![verifiable_credential];
+        let pre = to.create_presentation(credentials);
+
+        assert!(pre.is_ok());
+
+        let interim_presentation = pre.unwrap();
+
+        assert!(
+            interim_presentation.verifiable_credential[0]
+                .proof
+                .verification_method
+                .len()
+                > 0
+        );
+        assert!(
+            interim_presentation.verifiable_credential[0]
+                .proof
+                .proof_value
+                .len()
+                > 0
+        );
```

Comment:
```suggestion
        let interim_proof = &interim_presentation.verifiable_credential[0].proof;
        assert!(!interim_proof.verification_method.is_empty());
        assert!(!interim_proof.proof_value.is_empty());
```

Reply:
We're checking to see the inverse of `interim_proof.proof_value.is_empty()` here so we prepend the `!`